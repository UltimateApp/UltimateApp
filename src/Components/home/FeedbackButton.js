import React from 'react';
import { Alert } from 'react-native';
import * as MailComposer from 'expo-mail-composer';
import { MaterialCommunityIcons } from '@expo/vector-icons';

import I18n from '../../utils/i18n';
import theme from '../../styles/theme.style';
import HeaderButton from '../shared/HeaderButton';

export const FeedbackButton = (props) => {
  const feedback = () => {
    Alert.alert(
      I18n.t('feedback.alert.title'),
      I18n.t('feedback.alert.content'),
      [
        { text: I18n.t('feedback.alert.cancel'), style: 'cancel' },
        { text: I18n.t('feedback.alert.cta'), onPress: sendEmailAsync },
      ],
      { cancelable: true },
    );
  };

  const sendEmailAsync = () => {
    const result = MailComposer.composeAsync({
      recipients: ['ultimate.discin@gmail.com'],
      subject: I18n.t('feedback.subject'),
      body: '',
    });

    alert(result.status);
  };

  return <HeaderButton icon="email-outline" onPress={feedback} />;
};

export default FeedbackButton;
