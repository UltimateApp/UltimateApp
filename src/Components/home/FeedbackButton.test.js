import React from 'react';
import { create, act } from 'react-test-renderer';

import FeedbackButton from './FeedbackButton';

describe('<FeedbackButton />', () => {
  it('renders correctly', async () => {
    const tree = create(<FeedbackButton />).toJSON();
    await act(async () => expect(tree).toMatchSnapshot());
  });
});
