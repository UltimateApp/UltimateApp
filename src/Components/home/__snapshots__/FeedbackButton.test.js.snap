// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FeedbackButton /> renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "marginRight": 20,
      "opacity": 1,
    }
  }
>
  <Text />
</View>
`;
